app-ios/Sources/ContributorFeature/ContributorView.swift

```diff
@@ -1,27 +1,119 @@
 import SwiftUI
 import ComposableArchitecture
+import Model
+import Assets
+import Theme
+import NukeUI
 
 public struct ContributorState: Equatable {
-    public init() {}
+    public var contributors: [Contributor]
+
+    public init(contributors: [Contributor] = []) {
+        self.contributors = contributors
+    }
 }
 
-public enum ContributorAction {}
+public enum ContributorAction {
+    case refresh
+    case refreshResponse(TaskResult<[Contributor]>)
+}
 public struct ContributorEnvironment {
-    public init() {}
+    public let contributorsRepository: ContributorsRepository
+
+    public init(contributorsRepository: ContributorsRepository) {
+        self.contributorsRepository = contributorsRepository
+    }
 }
-public let contributorReducer = Reducer<ContributorState, ContributorAction, ContributorEnvironment> { _, _, _ in
-    return .none
+public let contributorReducer = Reducer<ContributorState, ContributorAction, ContributorEnvironment> { state, action, environment in
+    switch action {
+    case .refresh:
+        return .run { @MainActor subscriber in
+            for try await result: [Contributor] in environment.contributorsRepository.contributors().stream() {
+                await subscriber.send(
+                    .refreshResponse(
+                        TaskResult {
+                            result
+                        }
+                    )
+                )
+            }
+        }
+        .receive(on: DispatchQueue.main.eraseToAnyScheduler())
+        .eraseToEffect()
+    case .refreshResponse(.success(let contributors)):
+        state.contributors = contributors
```

Comment:
when update state, bring back to previous displayed tab 🤔 (I could not fix this problem)